doc/devdocs/locks.rst

```diff
@@ -0,0 +1,128 @@
+.. currentmodule:: Base
+
+****************************************************
+Proper maintenance and care of multi-threading locks
+****************************************************
+
+The following strategies are used to ensure that the code is dead-lock free
+(generally by addressing the 4th Coffman condition: circular wait).
```

Comment:
Something is better than nothing, but would rather try to make what's in this repo as approachable as we can. The purpose of them is so people who don't understand things yet can learn more about them.